File: V2rayNG/app/src/main/java/com/neko/marquee/text/LibVersions.java
package com.neko.marquee.text;

import android.content.Context;
import android.content.Intent;
import android.util.AttributeSet;
import android.view.View;
import androidx.appcompat.widget.AppCompatTextView;
import com.v2ray.ang.util.SpeedtestUtil;

public class LibVersions extends AppCompatTextView {
    private String memekVersion;

    private void JupokInfoSlur() {
        StringBuilder sb = new StringBuilder();
        sb.append(SpeedtestUtil.INSTANCE.getLibVersion()).toString();
        setText(sb);
    }

    @Override
    public boolean isFocused() {
        return true;
    }

    public LibVersions(Context context) {
        super(context);
        JupokInfoSlur();
    }

    public LibVersions(Context context, AttributeSet attrs) {
        super(context, attrs);
        JupokInfoSlur();
    }

    public LibVersions(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        JupokInfoSlur();
    }
}

File: V2rayNG/app/src/main/java/com/neko/marquee/text/LibVersions.kt
package com.neko.marquee.text

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.neko.v2ray.handler.SpeedtestManager

class LibVersions(context: Context) : AppCompatTextView(context) {
    private var memekVersion: String? = null

    private fun JupokInfoSlur() {
        val sb = StringBuilder()
        sb.append(SpeedtestManager.getLibVersion())
        text = sb.toString()
    }

    override fun isFocused(): Boolean {
        return true
    }

    constructor(context: Context, attrs: AttributeSet?) : this(context) {
        JupokInfoSlur()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : this(context, attrs) {
        JupokInfoSlur()
    }

    init {
        JupokInfoSlur()
    }
}

